snapcraft/internal/pluginhandler.py

```diff
@@ -874,3 +881,23 @@ def check_for_collisions(parts):
         # And add our files to the list
         parts_files[part.name] = {'files': part_files,
                                   'installdir': part.installdir}
+
+
+def _combine_filesets(fileset_1, fileset_2):
+    """Combine filesets if the first is an explicit or implicit wildcard."""
+
+    to_combine = False
+    # combine if fileset_1 has a wildcard
+    # XXX: should this only be a single wildcard and possibly excludes?
+    if '*' in fileset_1:
+        to_combine = True
+        fileset_1.remove('*')
+
+    # combine if fileset_1 is only excludes
+    if set([x[0] for x in fileset_1]) == set('-'):
```

Comment:
I'm not sure I understand your question.  Do you mean simply pulling the first set(...) out into a variable or something else?